### PR TITLE
Fix AutoEat crash

### PR DESCRIPTION
**Pull request by ReinforceZwei**
#1060 
Also added error handling for some ChatBot API

### CODE REVIEW

File: MinecraftClient/ChatBots/AutoEat.cs [hunk 1]
```diff
@@ -34,6 +34,7 @@ public override void Update()
 
         public override void OnHealthUpdate(float health, int food)
         {
+            if (health >= 0) return; // player dead
```
Comment (ORelio) [inline MinecraftClient/ChatBots/AutoEat.cs:37]: health <= 0 instead, right?

File: MinecraftClient/McTcpClient.cs [hunk 2]
```diff
@@ -1615,7 +1615,20 @@ public void OnUpdateHealth(float health, int food)
...
+            {
+                try
+                {
+                    bot.OnHealthUpdate(health, food);
+                }
+                catch (Exception e)
+                {
+                    if (!(e is ThreadAbortException))
+                    {
+                        ConsoleIO.WriteLogLine("OnPlayerProperty: Got error from " + bot.ToString() + ": " + e.ToString());
```
Comment (ORelio) [inline MinecraftClient/McTcpClient.cs:1627]: Thanks for adding an error handler here. I thing the message should be `OnHealthUpdate: Got error from [...]`.

File: MinecraftClient/McTcpClient.cs [hunk 3]
```diff
@@ -1629,7 +1642,20 @@ public void OnSetExperience(float Experiencebar, int Level, int TotalExperience)
...
+            {
+                try
+                {
+                    bot.OnSetExperience(Experiencebar, Level, TotalExperience);
+                }
+                catch (Exception e)
+                {
+                    if (!(e is ThreadAbortException))
+                    {
+                        ConsoleIO.WriteLogLine("OnPlayerProperty: Got error from " + bot.ToString() + ": " + e.ToString());
```
Comment (ORelio) [inline MinecraftClient/McTcpClient.cs:1654]: `OnSetExperience: Got error from [...]`

File: MinecraftClient/McTcpClient.cs [hunk 4]
```diff
@@ -1641,7 +1667,20 @@ public void OnSetExperience(float Experiencebar, int Level, int TotalExperience)
...
+            {
+                try
+                {
+                    bot.OnExplosion(location, strength, affectedBlocks);
+                }
+                catch (Exception e)
+                {
+                    if (!(e is ThreadAbortException))
+                    {
+                        ConsoleIO.WriteLogLine("OnPlayerProperty: Got error from " + bot.ToString() + ": " + e.ToString());
```
Comment (ORelio) [inline MinecraftClient/McTcpClient.cs:1679]: `OnExplosion: Got error from [...]`

File: MinecraftClient/McTcpClient.cs [hunk 5]
```diff
@@ -1656,7 +1695,20 @@ public void OnLatencyUpdate(Guid uuid, int latency)
...
+                {
+                    try
+                    {
+                        bot.OnLatencyUpdate(playerName, uuid, latency);
+                    }
+                    catch (Exception e)
+                    {
+                        if (!(e is ThreadAbortException))
+                        {
+                            ConsoleIO.WriteLogLine("OnPlayerProperty: Got error from " + bot.ToString() + ": " + e.ToString());
```
Comment (ORelio) [inline MinecraftClient/McTcpClient.cs:1707]: `OnLatencyUpdate: Got error from [...]`

File: MinecraftClient/McTcpClient.cs [hunk 6]
```diff
@@ -1669,7 +1721,20 @@ public void OnLatencyUpdate(Guid uuid, int latency)
...
+            {
+                try
+                {
+                    bot.OnHeldItemChange(slot);
+                }
+                catch (Exception e)
+                {
+                    if (!(e is ThreadAbortException))
+                    {
+                        ConsoleIO.WriteLogLine("OnPlayerProperty: Got error from " + bot.ToString() + ": " + e.ToString());
```
Comment (ORelio) [inline MinecraftClient/McTcpClient.cs:1733]: `OnHeldItemChange: Got error from [...]`

**Comment by ReinforceZwei** [comment]
Oops my carelessness. Fixed.

**Pull request merged**